app/middlewares/validations/albums.js

```diff
@@ -0,0 +1,17 @@
+const { validationError } = require('../../errors'),
+  logger = require('../../logger'),
+  { getAlbumById } = require('../../services/typicode');
+
+exports.validateId = (req, res, next) => {
+  if (isNaN(req.params.id) || req.params.id < 1) {
+    return next(validationError('album id must be a positive integer'));
+  }
+  const albumId = parseInt(req.params.id);
+
+  return getAlbumById(albumId)
```

Comment:
We don't often call services or models in middlewares, mostly because a model or service call is usually part of the main logic of the endpoint, but in this case it works very well since its part of the validation and does not affect the logic of the controller. 
As a headsup before I forget, if you ever need to pass data between middlewares, you can store it in the req object.

Reply:
Great! Good to know.
Is there any convention on how to set up the req at the end of the request? Maybe deleting all the things we add.

Reply:
I'm changing the name of the migrations. Is there any conventions on that?

Reply:
Not a convention per-say but you always want to add the type of the operation (rename, change, create, delete, etc) and the involved table. If the change is more specific, like only changing one column, you should add the name of said column as well.